pkg/sdk/session.go

```diff
@@ -0,0 +1,106 @@
+package sdk
+
+import (
+	"context"
+	"log"
+	"net/url"
+
+	sessiontypes "github.com/pokt-network/poktroll/x/session/types"
+	sharedtypes "github.com/pokt-network/poktroll/x/shared/types"
+)
+
+// sessionSuppliers is the structure that represents a session's end block height
+// and its matching suppliers.
+type sessionSuppliers struct {
+	// SessionEndBlockHeight is the session's end block height that helps
+	// determine if the session is still valid without looking into SupplierEndpoints slice.
+	SessionEndBlockHeight int64
+	SuppliersEndpoints    []*SupplierEndpoint
```

Comment:
Why not store the whole Supplier struct here?

Reply:
It was storing the whole `Supplier` struct in the first iteration. But this would make us flatten it each time `GetSessionSupplierEndpoints` is called.

See the reply below as to why we have to flatten the `Session` struct into `SupplierEndpoint`

Reply:
#PUC on why we need to flatten this